client/rest-high-level/src/main/java/org/elasticsearch/client/ml/inference/trainedmodel/ensemble/Ensemble.java

```diff
@@ -0,0 +1,188 @@
+    }
+
+    private final List<String> featureNames;
+    private final List<TrainedModel> models;
+    private final OutputAggregator outputAggregator;
+    private final TargetType targetType;
+    private final List<String> classificationLabels;
+
+    Ensemble(List<String> featureNames,
+             List<TrainedModel> models,
+             @Nullable OutputAggregator outputAggregator,
+             TargetType targetType,
+             @Nullable List<String> classificationLabels) {
+        this.featureNames = featureNames;
+        this.models = models;
+        this.outputAggregator = outputAggregator;
+        this.targetType = targetType;
+        this.classificationLabels = classificationLabels;
+    }
+
+    @Override
+    public List<String> getFeatureNames() {
+        return featureNames;
+    }
+
+    @Override
+    public String getName() {
+        return NAME;
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, ToXContent.Params params) throws IOException {
+        builder.startObject();
+        if (featureNames != null) {
+            builder.field(FEATURE_NAMES.getPreferredName(), featureNames);
+        }
+        if (models != null) {
+            NamedXContentObjectHelper.writeNamedObjects(builder, params, true, TRAINED_MODELS.getPreferredName(), models);
+        }
+        if (outputAggregator != null) {
+            NamedXContentObjectHelper.writeNamedObjects(builder,
+                params,
+                false,
+                AGGREGATE_OUTPUT.getPreferredName(),
+                Collections.singletonList(outputAggregator));
+        }
+        if (targetType != null) {
+            builder.field(TARGET_TYPE.getPreferredName(), targetType);
+        }
+        if (classificationLabels != null) {
+            builder.field(CLASSIFICATION_LABELS.getPreferredName(), classificationLabels);
+        }
+        builder.endObject();
+        return builder;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        Ensemble that = (Ensemble) o;
+        return Objects.equals(featureNames, that.featureNames)
+            && Objects.equals(models, that.models)
+            && Objects.equals(targetType, that.targetType)
+            && Objects.equals(classificationLabels, that.classificationLabels)
+            && Objects.equals(outputAggregator, that.outputAggregator);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(featureNames, models, outputAggregator, classificationLabels, targetType);
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static class Builder {
+        List<String> featureNames;
+        List<TrainedModel> trainedModels;
+        OutputAggregator outputAggregator;
+        TargetType targetType;
+        List<String> classificationLabels;
```

Comment:
These variables are not `private`.  It's not a big problem at the moment as the setters don't do any critical processing.  But in the future if a setter did anything extra then there would be a way to bypass it.  So unless there's a really good reason not to I'd make these `private`.

Reply:
They should totally be `private`. Text editing error.